File: 백준/Silver/1920. 수 찾기/수 찾기.py
import sys
read_line = sys.stdin.readline

n = int(read_line())
nums = set(map(int, read_line().split()))

m = int(read_line())
query = list(map(int, read_line().split()))

for num in query:
    print(1 if num in nums else 0)

File: 백준/Silver/1920. 수 찾기/수 찾기.java
import java.util.*;
import java.io.*;

class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        
        int n = Integer.parseInt(br.readLine());
        HashSet<Integer> nums = new HashSet<>();
        StringTokenizer st = new StringTokenizer(br.readLine());
        while(n-- > 0) {
            nums.add(Integer.parseInt(st.nextToken()));
        }
        
        int m = Integer.parseInt(br.readLine());
        st = new StringTokenizer(br.readLine());
        while(m-- > 0) {
            int num = Integer.parseInt(st.nextToken());
            System.out.println(nums.contains(num) ? 1 : 0);
        }
        br.close();
    }
}